internal/controller/ledger/controller_default.go

```diff
@@ -405,6 +405,13 @@ func (ctrl *DefaultController) revertTransaction(ctx context.Context, store Stor
 				balances[posting.Source][posting.Asset],
 				big.NewInt(0).Neg(posting.Amount),
 			)
+			if _, ok := balances[posting.Destination]; ok {
+				// if destination is also a source in some posting, since balances should only contain posting sources
+				balances[posting.Destination][posting.Asset] = balances[posting.Destination][posting.Asset].Add(
+					balances[posting.Destination][posting.Asset],
+					posting.Amount,
+				)
+			}
```

Comment:
I think this could be simplified this way (I assume Asset is a big.Int)
```suggestion
			if postingDestination, ok := balances[posting.Destination]; ok {
				// if destination is also a source in some posting, since balances should only contain posting sources
				
				postingDestination[posting.Asset].Add(
					postingDestination[posting.Asset],
					posting.Amount,
				)
			}
```

Reply:
oh, it's merged
nevermind